### PR TITLE
[ao] Add method in ModelReport to generate visualizer

### PR DESCRIPTION
**Pull request by vspenubarthi**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #81673
* #81647
* __->__ #81589
* #81523

Summary: We created a ModelReportVisualizer class, and the primary
way it is envisioned that it is accessed is:

```
model_report_visualizer = model_reporter.generate_visualizer()
```

This method only works after reports have been generated and it takes in
the generated reports and reformats them to be ordered by module, into
the format required by the ModelReportVisualization. It then generates
the visualizer instance and returns that to the user.

Test Plan: python test/test_quantization.py TestFxModelReportClass.test_generate_visualizer

Reviewers:

Subscribers:

Tasks:

Tags:

### CODE REVIEW

File: test/quantization/fx/test_model_report_fx.py [hunk 1]
```diff
@@ -1033,7 +1034,7 @@ def test_generate_report(self):
 
             # prepare and callibrate two different instances of same model
             # prepare the model
-            example_input = torch.randn(1, 3, 3, 3)
+            example_input = model_full.get_example_inputs()[0]
```
Comment (jerryzh168) [inline test/quantization/fx/test_model_report_fx.py:1037]: this should be a tuple of inputs (e.g. example_inputs = (torch.randn(1, 3, 3, 3),))

File: torch/ao/quantization/fx/_model_report/model_report.py [hunk 2]
```diff
@@ -275,5 +298,78 @@ def generate_model_report(
...
+        by the generate_model_report() method.
+
+        Returns the generated ModelReportVisualizer instance initialized
+
+        Note:
+            Throws exception if attempt to get visualizers without generating report
+        """
+        # check if user has generated reports at least once
+        if len(self._generated_reports) == 0:
+            raise Exception("You need to generate reports before you generate visualizers")
```
Comment (andrewor14) [inline torch/ao/quantization/fx/_model_report/model_report.py:367]: nit: This sounds a bit too colloquial? Maybe something like "Unable to generate visualizers without first generating reports"

File: torch/ao/quantization/fx/_model_report/model_report.py [hunk 3]
```diff
@@ -275,5 +298,78 @@ def generate_model_report(
...
+
+            for module_fqn in module_info:
+                # check if already in our accumulation dict
+                if module_fqn in module_fqns_to_features:
+                    # we merge all the features together
+                    new_info: Dict = module_info[module_fqn]
+                    present_info: Dict = module_fqns_to_features[module_fqn]
+
+                    # merge them together into the new unioned dict
+                    # same features keys -> same info, so okay if override
```
Comment (andrewor14) [inline torch/ao/quantization/fx/_model_report/model_report.py:336]: Maybe we should add a safety check to make sure the infos are in fact the same for the same keys

**Review by andrewor14 — APPROVED** [review]

File: test/quantization/fx/test_model_report_fx.py (hunk 1 above)
Comment (vspenubarthi) [reply test/quantization/fx/test_model_report_fx.py:1037]: is a tuple of inputs where we get it from the actual class

File: torch/ao/quantization/fx/_model_report/model_report.py (hunk 3 above)
Comment (vspenubarthi) [reply torch/ao/quantization/fx/_model_report/model_report.py:336]: Added

**Pull request closed without merging**